tests/ZiraLink.Api.Application.UnitTests/Services/ProjectServiceTests.cs

```diff
@@ -102,6 +103,23 @@ public async Task CreateProject_WhenCustomerIsNotValid_ShouldBeFailed(long custo
             var exception = await Assert.ThrowsAsync<NotFoundException>(() => projectService.CreateAsync(customerId, title, domainType, domain, internalUrl, state, CancellationToken.None));
             Assert.Equal("Customer", exception.Message);
         }
+        
+        [Theory]
+        [InlineData(1, "TestTitle", DomainType.Default, "TestDomain2", "https://google.com", ProjectState.Active)]
+        public async Task CreateProject_WhenInternalUrlIsNotValid_ShouldBeFailed(long customerId, string title, DomainType domainType, string domain, string internalUrl, ProjectState state)
+        {  
+            Mock<ILogger<ProjectService>> mockILoggerProjectService = new Mock<ILogger<ProjectService>>();
+            Mock<IBus> mockIBus = new Mock<IBus>();
+            Mock<IHttpTools> mockIHttpTools = new Mock<IHttpTools>();
+            mockIHttpTools.Setup(p => p.CheckDomainExists(It.IsAny<string>())).ReturnsAsync(false);
+
+            ProjectService projectService = new ProjectService(mockILoggerProjectService.Object, TestTools.AppMemoryDbContext, mockIBus.Object, mockIHttpTools.Object);
+
+            var exception = await Assert.ThrowsAsync<ApplicationException>(() => projectService.CreateAsync(customerId, title, domainType, domain, internalUrl, state, CancellationToken.None));
+
+            mockIHttpTools.Verify(p => p.CheckDomainExists(internalUrl), Times.Once());
+            Assert.Equal("Public domain is not allowed", exception.Message);
```

Comment:
Message could be updated ;)

Reply:
Removed.

Reply:
It was actually correct! I was just confused. an improvement could be made to the method's name, what you could do is you could change the method's name to something like `CreateProject_WhenInternalUrlIsAccessableFromInternet_ShouldBeFailed` or `CreateProject_WhenInternalUrlIsAPublicUrl_ShouldBeFailed`